
<launch>

  <arg name="robot"/>
  <arg name="simulated" default='true'/>
  <arg name="use_rik" default='true' />

  <!-- UR3e -->
  <group if="$(eval arg('robot') == 'ur3e')">
    <rosparam param="robot">'ur3e'</rosparam>
    <rosparam param="arm_move_group">'manipulator'</rosparam>
    <rosparam param="gripper_move_group">'gripper'</rosparam>
    <rosparam param="gripper_type">'robotiq-85'</rosparam>

    <include file="$(find iter_app)/launch/robots/ur3e.launch">
      <arg name="simulated" value="$(arg simulated )"/>
      <arg name="use_rik" value="$(arg use_rik)" />
    </include>
  </group>

  <!-- UR5 -->
  <group if="$(eval arg('robot') == 'ur5')">
    <rosparam param="robot">'ur5'</rosparam>
    <rosparam param="arm_move_group">'manipulator'</rosparam>
    <rosparam param="gripper_move_group">'gripper'</rosparam>
    <rosparam param="gripper_type">'robotiq-85'</rosparam>

    <include file="$(find iter_app)/launch/robots/ur5.launch">
      <arg name="simulated" value="$(arg simulated )"/>
      <arg name="use_rik" value="$(arg use_rik)" />
    </include>
  </group>

  <!-- Kinova Mico 3 Finger-->
  <group if="$(eval arg('robot') == 'mico-3')">
    <rosparam param="robot">'mico-3'</rosparam>
    <rosparam param="arm_move_group">'arm'</rosparam>
    <rosparam param="gripper_move_group">'gripper'</rosparam>
    <rosparam param="gripper_type">'mico-3'</rosparam>

    <include file="$(find iter_app)/launch/robots/mico-3.launch">
      <arg name="simulated" value="$(arg simulated )"/>
      <arg name="use_rik" value="$(arg use_rik)" />
    </include>
  </group>

  <!-- Kinova Mico 2 Finger-->
  <group if="$(eval arg('robot') == 'mico-2')">
    <rosparam param="robot">'mico-2'</rosparam>
    <rosparam param="arm_move_group">'arm'</rosparam>
    <rosparam param="gripper_move_group">'gripper'</rosparam>
    <rosparam param="gripper_type">'mico-2'</rosparam>

    <include file="$(find iter_app)/launch/robots/mico-2.launch">
      <arg name="simulated" value="$(arg simulated )"/>
      <arg name="use_rik" value="$(arg use_rik)" />
    </include>
  </group>

  <!-- Kinova Mico + Robotiq 85 -->
  <group if="$(eval arg('robot') == 'mico-robotiq85')">
    <rosparam param="robot">'mico-robotiq85'</rosparam>
    <rosparam param="arm_move_group">'arm'</rosparam>
    <rosparam param="gripper_move_group">'gripper'</rosparam>
    <rosparam param="gripper_type">'robotiq-85'</rosparam>

    <include file="$(find iter_app)/launch/robots/mico-robotiq85.launch">
      <arg name="simulated" value="$(arg simulated )"/>
      <arg name="use_rik" value="$(arg use_rik)" />
    </include>
  </group>

</launch>
